dockerspawner/dockerspawner.py

```diff
@@ -360,10 +372,20 @@ def start(self, image=None, extra_create_kwargs=None,
         # start the container
         yield self.docker('start', self.container_id, **start_kwargs)
 
-        # get the public-facing ip, port
-        resp = yield self.docker('port', self.container_id, 8888)
-        self.user.server.ip = self.container_ip
-        self.user.server.port = resp[0]['HostPort']
+        ip, port = yield from self.get_ip_and_port()
+        self.user.server.ip = ip
+        self.user.server.port = port
+
+    def get_ip_and_port(self):
+        if self.use_internal_ip:
+            inspect_resp = yield self.docker('inspect_container', self.container_id)
+            ip = inspect_resp['NetworkSettings']['IPAddress']
```

Comment:
Should be possible, but just to be sure, as I don't know the code very well: Does every user get its own spawner? As those are the IPs for the user servers/containers so every user gets a different IP.